server/commands.go

```diff
@@ -387,3 +388,30 @@ func getDynamicConfigValues(input []string) (map[dynamicconfig.Key][]dynamicconf
 	}
 	return ret, nil
 }
+
+func availablePort(port int) int {
+	i := 0
+
+	for isPortTaken("127.0.0.1", port+i) {
+		i += 1
+	}
+
+	return port + i
+}
+
+func isPortTaken(host string, port int) bool {
+	timeout := 2 * time.Second
+	target := fmt.Sprintf("%s:%d", host, port)
+
+	conn, err := net.DialTimeout("tcp", target, timeout)
+	if err != nil {
+		return false
+	}
+
+	if conn != nil {
+		_ = conn.Close()
+		return true
+	}
+
+	return false
+}
```

Comment:
I've added this code here for the sake of simplicity. I questioned whether it makes sense to move it to the [PortProvider](https://github.com/temporalio/cli/blob/main/server/config/freeport.go#L40) struct. Alternatively, if there's a better place for it, please let me know. What do all of you think?

Reply:
I agree changing PortProvider, or just using it outright, makes sense. (I think it already has some functionality for picking a free port, so you don't have to reimplement this.)